management/WFCORE-5489_override_attribute_value_from_env_var.adoc

```diff
@@ -0,0 +1,138 @@
+= WFCORE-5489 - As a developer, I want to override management attribute values using environment variables
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+WildFly on the cloud relies on container environment variables to customize its configuration to the container platform.
+
+It is difficult to change the value of a management attribute from the standalone configuration if the attribute does not provide a default value with an expression that can be resolved at runtime.
+
+For example, the user wants to update the `proxy-address-forwarding` attribute on the `/subsystem=undertow/server=default-server/http-listener=default` resource.
+
+It is not possible to achieve this with our existing configuration as the value is not specified in standalone.xml and defaults to `false`.
+If we want to allow our users to change this attribute, we must explicitely set its value with an expression that defaults to its default value:
+
+[source]
+----
+/subsystem=undertow/server=default-server/http-listener=default:write-attribute(name=proxy-address-forwarding, value=${undertow.default.http-listener.proxy-address-forwarding:false)
+----
+
+[NOTE]
+====
+This is an actual use case as we are changing the value of this attribute when a Bootable jar is targeting a cloud deployment in https://github.com/wildfly-extras/wildfly-jar-maven-plugin/blob/master/plugin/src/main/resources/org/wildfly/plugins/bootablejar/maven/cloud/openshift-undertow-script.cli 
+====
+
+With that change, the user can now use the expression resolution to change the attribute value.
+
+However this implies for *any* attribute that can be modified by the user at deployment time to explicity set its value to a resolvable expression with a default.
+This is not achievable while keeping our configuration readable and maintainable.
+
+Instead this RFE proposes to introduce a generic mechanism to let user override *any* management attribute using environment variables.
```

Comment:
Would these overridden values coming from the environment values be persisted on the server configuration files?

I'm not sure how this is going to be implemented, but keep in mind that the configuration scripts use bash to read existing configurations from the server configuration files. I would say any overrides will be done always after configuring the server via shell scripts because if they are done before without being persisted in the server configuration files, the script execution will never take them into account. Maybe this situation deserves a hard requirement describing it.

Reply:
The idea behind this feature is to affect the in-memory management model without affecting its storage.

The main use case is for cloud deployment where the configuration is read once when the pod is started.
However, I'll clarify the requirements when we use it on bare metal.

Reply:
@yersan I've clarified the requirement. Could you please review it?

Reply:
Thanks @jmesnil , yes, I think the hard requirement about the cloud deployment makes it more clear.

The launch scripts will read the initial server configuration without taking into account values overridden by these env variables. The environment variables will override the configuration generated by the launch scripts.